packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -18,6 +19,29 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertComponentElement(this.firstChild, { content: 'hello' });
   }
 
+  // Note this functionality seems weird
```

Comment:
In `{{foo-bar id=blah}}`, we evaluate the value of `blah` and "freeze" it as `elementId` in the component instance.

`elementId` can't be changed, if it did change then any actions/events would stop working. The view registry is keyed on the `elementId`, and is used to look up action/event targets. So it is definitely possible to make changing the `id`/`elementId` work, but it would be a new feature.

We should ensure that a test also exists in this file for:

```
let fooBarInstance;
this.registerComponent('foo-bar', {
  ComponentClass: Component.extend({
    elementId: 'blahzorz',
    init() {
      this._super(...arguments);
      fooBarInstance = this;
    }
  }),

  template: `{{elementId}}`
});

this.render('{{foo-bar}}');

this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { id: 'blahzorz' }, content: 'blahzorz'});

this.runtask(() => {
  expectAssertion(() => {
    fooBarInstance.set('elementId', 'herpyderpy');
  }, /can not change elementId/);
});

this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { id: 'blahzorz' }, content: 'blahzorz'});
```